contracts/Forwarder.sol

```diff
@@ -50,15 +41,26 @@ contract Forwarder {
     }
 
     /**
-     * @notice Transfer the token balance held by the forwarder to the wallet.
-     * @param _wallet the wallet controlling the zkSync assets deposited by the forwarder
-     * @param _token the token to transfer
+     * @notice Transfer the ERC20 token balance held by the forwarder to the wallet.
+     * @param _wallet the wallet associated to the forwarder
+     * @param _token the ERC20 token to transfer
      */
-    function recoverToken(address _wallet, address _token) external onlyFactory {
+    function recoverERC20Token(address _wallet, address _token) external onlyFactory {
         IERC20 tokenContract = IERC20(_token);
         uint256 balance = tokenContract.balanceOf(address(this));
         tokenContract.safeTransfer(_wallet, balance);
     }
 
+    /**
+     * @notice Transfer an ERC721 token held by the forwarder to the wallet.
+     * @param _wallet the wallet associated to the forwarder
+     * @param _token the ERC721 token contract
+     * @param _id the ERC721 token to transfer
+     */
+    function recoverERC721Token(address _wallet, address _token, uint256 _id) external onlyFactory {
+        IERC721 tokenContract = IERC721(_token);
+        tokenContract.safeTransferFrom(address(this), _wallet, _id);
```

Comment:
Note that this uses `SafeERC20.safeTransferFrom` but the intend here is to do a `IERC721.transferFrom` or `IERC721.safeTransferFrom` (the latter including a call to smart contract recipient which is not dont in `SafeERC20.safeTransferFrom`). Let's maybe change this line to `IERC721.transferFrom` to avoid confusion?